src/transformStateWithClones.js

```diff
@@ -8,6 +8,41 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const ACTION_TYPE_ADD_PROPERTIES = 'addProperties';
+  const ACTION_TYPE_REMOVE_PROPERTIES = 'removeProperties';
+  const ACTION_TYPE_CLEAR = 'clear';
+
+  let prevObject = { ...state };
+  const objHistory = [];
+
+  for (const obj of actions) {
```

Comment:
Also use destruction here for `action`
```suggestion
  for (const action of actions) {
  
  const { type, ........ } = action;
```